sdk/core/core-http/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -112,6 +112,7 @@ export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
       // Waiting for the next refresh only if the cache is unable to retrieve the access token,
       // which means that it has expired, or it has never been set.
       accessToken = await this.tokenRefresher.refresh(options);
+      this.tokenCache.setCachedToken(accessToken);
```

Comment:
I feel like I should have spotted this in the pervious review 🤦‍♂️

Reply:
The test and the recorders were screaming me that something was wrong and I didn't see it. I feel the burn.